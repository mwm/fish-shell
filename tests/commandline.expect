# vim: set filetype=expect:

spawn $fish

expect_prompt

send_line "bind '~' 'handle_tilde'"
expect_prompt

# printing the current buffer should not remove quoting
send_line "function handle_tilde; echo; echo '@GUARD:1@'; commandline -b; echo '@/GUARD:1@'; commandline -b ''; end"
expect_prompt
send_line {echo \en one "two three" four'five six'{7} 'eight~}
expect_prompt -re {\r\n@GUARD:1@\r\n(.*)\r\n@/GUARD:1@\r\n} {
    puts "a:"
    puts $expect_out(1,string)
} unmatched {
    abort "Couldn't find guard 1"
}

# printing the buffer with -o should remove quoting
send_line "function handle_tilde; echo; echo '@GUARD:2@'; commandline -bo; echo '@/GUARD:2@'; commandline -b ''; end"
expect_prompt
send_line {echo one "two three" four'five six'{7} 'eight~}
expect_prompt -re {\r\n@GUARD:2@\r\n(.*)\r\n@/GUARD:2@\r\n} {
    puts "b:"
    puts [string map {\r {}} $expect_out(1,string)]
} unmatched {
    abort "Couldn't find guard 2"
}
