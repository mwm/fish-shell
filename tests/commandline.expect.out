a:
echo \en one "two three" four'five six'{7} 'eight
b:
echo
one
two three
fourfive six{7}
eight
